### PR TITLE
Upload snapshots md5 as well

### LINKED ISSUE #476: Upload snapshots md5/sha as well
After the snapshot is uploaded, it's checksum file should be uploaded as well to easy further integrity checks

### PR DESCRIPTION
**Pull request by IvayloPapazov**
close #476

### CODE REVIEW

File: ansible/mnesia_snapshot.yml [hunk 1]
```diff
@@ -106,6 +133,21 @@
...
+        bucket: "{{ snapshots_bucket }}"
+        object: "{{ snapshot_hash_filename_latest }}"
+        src: "{{ snapshot_hash_path }}"
+        mode: put
+        permission: "public-read"
+        aws_access_key: "{{ lookup('env','AWS_ACCESS_KEY_ID') }}"
+        aws_secret_key: "{{ lookup('env','AWS_SECRET_ACCESS_KEY') }}"
+        security_token: "{{ lookup('env','AWS_SESSION_TOKEN') }}"
+      when:
+        - snapshot_file.stat is defined
```
Comment (venimus) [inline ansible/mnesia_snapshot.yml]: snapshot_file.stat.readable

File: ansible/mnesia_snapshot.yml [hunk 2]
```diff
@@ -91,6 +94,30 @@
...
+      tags: [upload]
+
+    - name: Upload snapshot hash to S3 bucket ({{ snapshots_bucket }}/{{ snapshot_hash_filename }})
+      aws_s3:
+        bucket: "{{ snapshots_bucket }}"
+        object: "{{ snapshot_hash_filename }}"
+        src: "{{ snapshot_hash_path }}"
+        mode: put
+        permission: "public-read"
+        aws_access_key: "{{ lookup('env','AWS_ACCESS_KEY_ID') }}"
```
Comment (venimus) [inline ansible/mnesia_snapshot.yml]: predefine these in the beginning

File: ansible/mnesia_snapshot.yml [hunk 3]
```diff
@@ -106,6 +133,21 @@
         - snapshot_file.stat.readable
       tags: [latest]
 
+    - name: Upload backup hash to S3 bucket latest
+      aws_s3:
+        bucket: "{{ snapshots_bucket }}"
+        object: "{{ snapshot_hash_filename_latest }}"
```
Comment (venimus) [inline ansible/mnesia_snapshot.yml]: i don't see where this is defined

File: ansible/mnesia_snapshot.yml [hunk 4]
```diff
@@ -9,6 +9,8 @@
     snapshot_suffix: snapshot
     snapshot_filename: "mnesia_{{ env }}_v-{{ db_version }}_{{ ansible_ssh_host|default(inventory_hostname) }}_{{ snapshot_suffix }}.tgz"
     snapshot_path: "{{ snapshots_dir|default('/tmp/snapshots') }}/{{ snapshot_filename }}"
+    snapshot_hash_filename: "{{ snapshot_filename }}.md5"
+    snapshot_hash_path: "{{ snapshots_dir|default('/tmp/snapshots') }}/{{ snapshot_hash_filename }}"
```
Comment (venimus) [inline ansible/mnesia_snapshot.yml]: IMHO this should not include the filename

Comment (IvayloPapazov) [reply ansible/mnesia_snapshot.yml]: df85214

File: ansible/mnesia_snapshot.yml (hunk 3 above)
Comment (IvayloPapazov) [reply ansible/mnesia_snapshot.yml]: df85214

File: ansible/mnesia_snapshot.yml (hunk 2 above)
Comment (IvayloPapazov) [reply ansible/mnesia_snapshot.yml]: df85214

File: ansible/mnesia_snapshot.yml (hunk 1 above)
Comment (IvayloPapazov) [reply ansible/mnesia_snapshot.yml]: df85214

File: ansible/mnesia_snapshot.yml [hunk 5]
```diff
@@ -83,9 +88,33 @@
...
-        security_token: "{{ lookup('env','AWS_SESSION_TOKEN') }}"
+        aws_access_key: "{{ aws_access_key }}"
+        aws_secret_key: "{{ aws_secret_key }}"
+        security_token: "{{ security_token }}"
+      when:
+        - snapshot_file.stat.readable is defined
+        - snapshot_file.stat.readable
+      tags: [upload]
+
+    - name: Add hash into ({{ snapshot_hash_path }})
```
Comment (venimus) [inline ansible/mnesia_snapshot.yml]: Generate a checksum file

File: ansible/mnesia_snapshot.yml [hunk 6]
```diff
@@ -83,9 +88,33 @@
...
+      copy:
+        dest: "{{ snapshot_hash_path }}"
+        content: "{{ snapshot_file.stat.checksum }}"
+      when:
+        - snapshot_file.stat.readable is defined
+        - snapshot_file.stat.readable
+      tags: [upload]
+
+    - name: Upload snapshot hash to S3 bucket ({{ snapshots_bucket }}/{{ snapshot_filename }}.md5)
```
Comment (venimus) [inline ansible/mnesia_snapshot.yml]: Upload checksum file

**Comment by venimus** [comment]
update the `clean snapshot` handler too

File: ansible/mnesia_snapshot.yml (hunk 5 above)
Comment (IvayloPapazov) [reply ansible/mnesia_snapshot.yml]: a9dd1b2

File: ansible/mnesia_snapshot.yml (hunk 6 above)
Comment (IvayloPapazov) [reply ansible/mnesia_snapshot.yml]: a9dd1b2

**Pull request merged**